l1-contracts/contracts/bridge/L1Nullifier.sol

```diff
@@ -486,24 +486,29 @@ contract L1Nullifier is IL1Nullifier, ReentrancyGuard, Ownable2StepUpgradeable,
     /// @return transferData The transfer data used to finalize withdawal.
     function _verifyWithdrawal(
         FinalizeL1DepositParams calldata _finalizeWithdrawalParams
-    ) internal view returns (bytes32 assetId, bytes memory transferData) {
-        (assetId, transferData) = _parseL2WithdrawalMessage(
+    ) internal view returns (bytes32 assetId, bytes memory transferData, WithdrawalType withdrawalType) {
+        (assetId, transferData, withdrawalType) = _parseL2WithdrawalMessage(
             _finalizeWithdrawalParams.chainId,
             _finalizeWithdrawalParams.message
         );
         L2Message memory l2ToL1Message;
         {
+            address l2Sender = _finalizeWithdrawalParams.l2Sender;
             bool baseTokenWithdrawal = (assetId == BRIDGE_HUB.baseTokenAssetId(_finalizeWithdrawalParams.chainId));
-            require(
-                /// @dev for legacy function calls we hardcode the sender as the L2AssetRouter as we don't know if it is
```

Comment:
note this here, this means later when you require that the l2Sender is something, that might be wrong. That why we had this || thing.